drivers/flash/Kconfig.simulator

```diff
@@ -0,0 +1,181 @@
+# Kconfig - Flash simulator config
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig FLASH_SIMULATOR
+	bool
+	prompt "Flash simulator"
+	default n
+	depends on FLASH
+	select STATS
+	select STATS_NAMES
+	select MULTITHREADING
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
+	help
+	  Enable the flash simulator.
+
+if FLASH_SIMULATOR
+
+config FLASH_SIMULATOR_BASE_ADDR
```

Comment:
Are we going to simulate internal flash (the one which can be mapped into CPU address space) ?
If no, this is not needed (flash will always start from 0). If yes, we should be able to detect accidental write to flash area (this can be done using MPU).